packages/shared/src/components/search/SearchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {
+  ReactElement,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { SearchPanelInput } from './SearchPanelInput';
+import { SearchProviderEnum } from '../../../graphql/search';
+import { SearchPanelContext } from './SearchPanelContext';
+import { minQueryLength, searchPanelGradientElementId } from './common';
+import { SearchPanelAction } from './SearchPanelAction';
+import { SearchPanelPostSuggestions } from './SearchPanelPostSuggestions';
+import { useFeature } from '../../GrowthBookProvider';
+import { feature } from '../../../lib/featureManagement';
+import { SearchExperiment } from '../../../lib/featureValues';
+import { FeedGradientBg } from '../../feeds/FeedGradientBg';
+import Portal from '../../tooltips/Portal';
+import SettingsContext from '../../../contexts/SettingsContext';
+import { useActions } from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+import { AnalyticsEvent } from '../../../lib/analytics';
+import { useAnalyticsContext } from '../../../contexts/AnalyticsContext';
+
+export type SearchPanelProps = {
+  className?: string;
+};
+
+export const SearchPanel = ({ className }: SearchPanelProps): ReactElement => {
+  const searchVersion = useFeature(feature.search);
+  const { sidebarExpanded } = useContext(SettingsContext);
+  const { trackEvent } = useAnalyticsContext();
+  const { completeAction, checkHasCompleted, isActionsFetched } = useActions();
+
+  const isTracked = useRef(false);
+  const shouldShowPulse =
+    isActionsFetched && !checkHasCompleted(ActionType.UsedSearchPanel);
+
+  const [state, setState] = useState(() => {
+    return {
+      provider: SearchProviderEnum.Posts,
+      query: '',
+      isActive: false,
+    };
+  });
+
+  const searchPanelContextValue = useMemo(() => {
+    return {
+      ...state,
+      setProvider: (provider: SearchProviderEnum) => {
+        setState((currentState) => {
+          return { ...currentState, provider };
+        });
+      },
+      setActive: (isActive: boolean) => {
+        setState((currentState) => {
+          return { ...currentState, isActive };
+        });
+      },
+    };
+  }, [state]);
+
+  const showDropdown = state.isActive && state.query.length >= minQueryLength;
+
+  const gradientContainer = useMemo(() => {
+    if (typeof state.isActive === 'undefined') {
+      return undefined;
+    }
+
+    if (typeof document === 'undefined') {
+      return undefined;
+    }
+
+    return (
+      document.getElementById(searchPanelGradientElementId) ||
+      document.createElement('div') // unmounted div to not show the gradient
+    );
+  }, [state.isActive]);
+
+  if (searchVersion !== SearchExperiment.V1) {
+    return null;
+  }
```

Comment:
Should we maybe ensure `SearchPanel` doesn't even get rendered?
Feels cheaper since above are quite some computations?

Reply:
yep